projects/core/src/organization/store/effects/budget.effect.spec.ts

```diff
@@ -78,16 +86,102 @@ describe('Budget Effects', () => {
     effects = TestBed.get(fromEffects.BudgetEffects as Type<
       fromEffects.BudgetEffects
     >);
+    budgetConnector = TestBed.get(BudgetConnector as Type<BudgetConnector>);
+    expected = null;
   });
 
   describe('loadBudget$', () => {
-    it('should return loadProductStart action if product not loaded', () => {
-      const action = new BudgetActions.LoadBudget({ uid, budgetCode });
-      const completion = new BudgetActions.LoadBudgetSuccess(budget);
+    it('should return LoadBudgetSuccess action', () => {
+      const action = new BudgetActions.LoadBudget({ userId, budgetCode });
+      const completion = new BudgetActions.LoadBudgetSuccess([budget]);
+      actions$ = hot('-a', { a: action });
+      expected = cold('-b', { b: completion });
+
+      expect(effects.loadBudget$).toBeObservable(expected);
+      expect(budgetConnector.get).toHaveBeenCalledWith(userId, budgetCode);
+    });
+
+    it('should return LoadBudgetFail action if budget not updated', () => {
+      budgetConnector.get = createSpy().and.returnValue(throwError(error));
+      const action = new BudgetActions.LoadBudget({ userId, budgetCode });
+      const completion = new BudgetActions.LoadBudgetFail(error);
```

Comment:
I think you need code in payload here. Is it provided in error returned from API?
Applies to every Fail action